src/Nethermind/Nethermind.Tools.Kute/JsonRpcMessageProvider/FileJsonRpcMessageProvider.cs

```diff
@@ -0,0 +1,38 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+namespace Nethermind.Tools.Kute.JsonRpcMessageProvider;
+
+class FileJsonRpcMessageProvider : IJsonRpcMessageProvider
+{
+    private readonly string _filePath;
+
+    public FileJsonRpcMessageProvider(string filePath)
+    {
+        _filePath = filePath;
+    }
+
+
+    public IAsyncEnumerable<string> Messages
+    {
+        get
+        {
+            var pathInfo = new FileInfo(_filePath);
+            if (pathInfo.Attributes.HasFlag(FileAttributes.Directory))
+            {
+                return Directory.GetFiles(_filePath)
+                    .Select(filePath => new FileInfo(filePath))
+                    .OrderBy(info => info.LastWriteTime)
```

Comment:
Do we want to provide a flag to decide how to sort in case of using a folder?

By default, using the `LastWriteTime` we ensure to execute them in the same order as in the EL.